apsuite/orbcorr/si_calc_bump_matrix_at_bendings.py

```diff
@@ -7,53 +7,95 @@
 import pyaccel
 from . import OrbitCorr
 
+MARKER_NAMES = {
+    'B1': 'B1_SRC',
+    'B2': 'B2_SRC',
+    'BC': 'mc',
+}
 
-def calc_matrices(minsingval=0.2, mb1idx=0, deltax=10e-6, n_bpms_out=3):
-    """."""
-    # NOTE: valid only for first B1 in sector, the one in subsec C1!
+BENDING_BPM_SEC_IDCS = {
+    'B1': [0, 1],
+    'B2': [2, 3],
+    'BC': [3, 4],
+}
+
+
+def _get_sec_bpm_indices(bending_type='B1'):
+    bdict = BENDING_BPM_SEC_IDCS
+    return bdict[bending_type][0], bdict[bending_type][1]
+
+
+def _get_closest_bpms_indices(bending_type='B1', bidx=0, n_bpms_out=2):
+    bpm1_sec_index, bpm2_sec_index = _get_sec_bpm_indices(bending_type)
+    idlist = np.arange(0, 160, 1)
+    idcs_ignore = list()
+    for i in np.arange(n_bpms_out):
+        idcs_ignore.append(idlist[bpm1_sec_index + 8*bidx - (i+1)])
+        idcs_ignore.append(idlist[bpm2_sec_index + 8*bidx + (i+1)])
+    idcs_ignore = np.array(idcs_ignore)
+    idcs_ignore = np.tile(idcs_ignore, 2)
+    idcs_ignore[n_bpms_out*2:] += 160
+    return idcs_ignore
+
+
+def _get_removed_corrs_indices(bending_type='B1', bidx=0):
+    ch_idcs = dict()
+    cv_idcs = dict()
+    ch_idcs['B1'] = [bidx*6 + 0, bidx*6 + 1]
+    cv_idcs['B1'] = [bidx*8 + 0, bidx*8 + 1]
+
+    ch_idcs['B2'] = [bidx*6 + 2]
+    cv_idcs['B2'] = [bidx*8 + 2, bidx*8 + 3]
 
-    bpm1_sec_index = 0
-    bpm2_sec_index = 1
+    ch_idcs['BC'] = []
+    cv_idcs['BC'] = []
+
+    return ch_idcs[bending_type], cv_idcs[bending_type]
+
+
+def calc_matrices(bending_type='B1', minsingval=0.2, bidx=0,
+                  deltax=10e-6, n_bpms_out=3):
+    """."""
+    # NOTE: valid only for bendings in subsectors C1, C2 or BC
+    bpm1_sec_index, bpm2_sec_index = _get_sec_bpm_indices(bending_type)
 
+    # Create accelerator and orbcorr
     mod = si.create_accelerator()
     mod.cavity_on = True
     orbcorr = OrbitCorr(mod, 'SI', use6dorb=True)
     orbcorr.params.enblrf = True
     orbcorr.params.tolerance = 1e-9
     orbcorr.params.minsingval = minsingval
 
-    mb1idx = max(min(mb1idx, 19), 0)
-    mb1i = pyaccel.lattice.find_indices(
-        orbcorr.respm.model, 'fam_name', 'B1_SRC')[2*mb1idx]
+    # Get source marker idx
+    bidx = max(min(bidx, 19), 0)
+    nr = 1 if bending_type == 'BC' else 2
+    mbend = pyaccel.lattice.find_indices(
+        orbcorr.respm.model, 'fam_name', MARKER_NAMES[bending_type])[nr*bidx]
 
     orb0 = orbcorr.get_orbit()
     kicks0 = orbcorr.get_kicks()
 
+    # Get BPM indices
     idcs = np.array(
         [bpm1_sec_index, bpm2_sec_index,
          160+bpm1_sec_index, 160+bpm2_sec_index])
-    idcs += 8*mb1idx
+    idcs += 8*bidx
 
     # remove corrs between BPMs
-    orbcorr.params.enbllistch[mb1idx*6 + 0] = False
-    orbcorr.params.enbllistch[mb1idx*6 + 1] = False
-    orbcorr.params.enbllistcv[mb1idx*8 + 0] = False
-    orbcorr.params.enbllistcv[mb1idx*8 + 1] = False
+    ch_idcs, cv_idcs = _get_removed_corrs_indices(bending_type, bidx)
```

Comment:
@Gabrielrezende-asc , maybe you could consider passing the object orbcor.params as an input and set the enabled list inside the method. What do you think?